tests/test_train.py

```diff
@@ -1,3 +1,83 @@
 import pytest,fastai
+from utils.fakes import *
+from utils.text import *
 
-#I'm all empty, please fill me with tests about the functions in train.py
\ No newline at end of file
+## filename: test_train.py
+## tests functions in train.py
+
+## Class Learner, see documentation:  https://docs.fast.ai/train.html
+ 
+## Model fitting methods
+
+## run: pytest tests/test_train.py (add -s for screenoutput )
+
+
+@pytest.fixture(scope="module")
+def learn():
+    
+    learn = fake_learner(50,50)
+   
+    return learn
+
+def test_fit(learn):
+
+    ## Test confirms learning rate and momentum are stable, see difference to test_fit_one_cycle
+    
+    learning_rate = 3e-3
+    weight_decay = 1e-2
+    eps = 4
+    
+    with CaptureStdout() as cs:  learn.fit(epochs=eps, lr=learning_rate, wd=weight_decay)
+   
+    lrs = list(learn.recorder.lrs)
+    prevlr = learning_rate
+    for lr in lrs:
+        if prevlr is not None: assert prevlr == lr
+        prevlr = lr
+    moms = list(learn.recorder.moms)
+    prevmom = None
+    for mom in moms:
+        if prevmom is not None: assert prevmom == mom
+        prevmom = mom
+   
+def test_fit_one_cycle(learn):
+
+    ## Test confirms expected behavior leaning rate and momentum
+    ## see graphical representation here: output cell 17 of, learn.sched.plot_lr() in 
+    ## https://github.com/sgugger/Deep-Learning/blob/master/Cyclical%20LR%20and%20momentums.ipynb
+
+    cycle_length = 4
+    lr = 3e-3
+
+    with CaptureStdout() as cs: learn.fit_one_cycle(cycle_length, lr)
+  
+    listlrs = list(learn.recorder.lrs)
+    listmoms = list(learn.recorder.moms) 
+    
+    ## eliminate the final low learning rates with corresponding momentum
```

Comment:
Why do you do this? It's normal we get fom init_lr to max_lr then 0 (not exactly 0 as we stop just before but almost).